### PR TITLE
Practice6 - final solution with mandatory requirements

### PR DESCRIPTION
**Pull request by stratopedarx**
Hello, please look into my PR.
At the moment, I switch to start working on the final project.
Thanks in advance.

### CODE REVIEW

File: .swiftlint.yml [hunk 1]
```diff
@@ -2,3 +2,4 @@ identifier_name:
   excluded:
     - id
     - dt
+    - q
```
Comment (vasilevkin) [inline .swiftlint.yml]: just a remark. clarity over brevity.

https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/CodingGuidelines/Articles/NamingBasics.html

File: practice_2/NewsApp/Presenter/NewsPresenter.swift [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        let lastApiDate = dbService.getDateFromDB()
+        let timeSinceLastApiUpdate = Int(now.timeIntervalSince(lastApiDate as Date))  // in seconds
+
+        if timeSinceLastApiUpdate > newsCacheExpired {
+            dbService.deleteAllArticlesFromDatabase()
+            getArticlesFromApi()
+        } else {
+            print("Get data from DB")
+            dbService.getArticlesFromDatabase(onComplete: { articlesDB in
+                if articlesDB.count == 0 {
```
Comment (vasilevkin) [inline practice_2/NewsApp/Presenter/NewsPresenter.swift]: do not compare to 0, use .isEmpty or something like that

File: practice_2/NewsApp/View/DetailsNewsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+            print("Could not create URL object for \(article.urlToImage)")
+        }
+        newsDescriptionTextView.text = article.description
+    }
+
+    private func getNewsUrl() -> URL? {
+        if let article = article {
+            return URL(string: article.url)
+        }
+        return URL(string: "https://www.google.com")
```
Comment (vasilevkin) [inline practice_2/NewsApp/View/DetailsNewsViewController.swift]: it's better to show some alert or warning, just not to confuse a user

File: practice_2/NewsApp/View/NewsCollectionViewController.swift [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        HamburgerMenu().triggerSideMenu()
+    }
+
+    // MARK: Sort alphabetically
+    @IBAction func sortAction(_ sender: UIBarButtonItem) {
+        guard let title = sender.title else { return }
+        if title == "up" {
+            self.presenter?.sortAlphabetically(inOrder: title)
+            sender.title = "down"
+            sender.image = UIImage(systemName: "arrow.down")
```
Comment (vasilevkin) [inline practice_2/NewsApp/View/NewsCollectionViewController.swift]: try move these system names to constants

**Review by vasilevkin — CHANGES_REQUESTED** [review]
Looks good, but there are some minor remarks

File: practice_2/NewsApp/View/DetailsNewsViewController.swift (hunk 3 above)
Comment (stratopedarx) [reply practice_2/NewsApp/View/DetailsNewsViewController.swift]: I agree with you. Fixed. Thanks.

File: practice_2/NewsApp/View/NewsCollectionViewController.swift (hunk 4 above)
Comment (stratopedarx) [reply practice_2/NewsApp/View/NewsCollectionViewController.swift]: Thank you. Done

File: practice_2/NewsApp/Presenter/NewsPresenter.swift (hunk 2 above)
Comment (stratopedarx) [reply practice_2/NewsApp/Presenter/NewsPresenter.swift]: Thank you.

File: .swiftlint.yml (hunk 1 above)
Comment (stratopedarx) [reply .swiftlint.yml]: Follow you. Thanks.

Comment (stratopedarx) [reply .swiftlint.yml]: and Fixed

**Comment by stratopedarx** [comment]
Thanks for your review.
Tried to fix all remarks.

**Review by vasilevkin — APPROVED** [review]
LGTM

**Pull request merged**